File: code_samples/python/v1.0.0/create_payment_pre_authorization.py
from tatrapayplus import TatrapayPlusClient
from tatrapayplus.models import *

client = TatrapayPlusClient(
    "your-client-id",
    "your-client-secret",
)

payment_data = InitiatePaymentRequest(
    base_payment=BasePayment(
        instructed_amount=Amount(
            amount_value=120.0,
            currency="EUR",
        ),
        end_to_end="ORDER123456",
    ),
    card_detail=CardDetail(
        is_pre_authorization=True,
        card_holder="Janko Hruska",
    ),
    bank_transfer=BankTransfer(),
)

response = client.create_payment(payment_data, "https://your-redirect-uri.com", "127.0.0.1")

File: code_samples/php/v2/create_payment_pre_authorization.php
<?php
use Tatrapayplus\TatrapayplusApiClient\Api\TatraPayPlusAPIApi;

$tatrapayplus_api = new TatraPayPlusAPIApi(
    "your-client-id",
    "your-client-secret",
);

$initiate_payment_request = new Tatrapayplus\TatrapayplusApiClient\Model\InitiatePaymentRequest([
    "base_payment" => new Tatrapayplus\TatrapayplusApiClient\Model\BasePayment([
        "instructed_amount" => new Tatrapayplus\TatrapayplusApiClient\Model\Amount([
            "amount_value" => 10.0,
            "currency" => "EUR",
        ]),
        "end_to_end" => new Tatrapayplus\TatrapayplusApiClient\Model\E2e([
            "variable_symbol" => "1",
            "specific_symbol" => "2",
            "constant_symbol" => "3",
        ]),
    ]),
    "bank_transfer" => new Tatrapayplus\TatrapayplusApiClient\Model\BankTransfer(),
    "card_detail" => new Tatrapayplus\TatrapayplusApiClient\Model\CardDetail([
        "card_holder" => "Janko Hrasko",
        "is_pre_authorization" => true,
    ]),
]);

$response = $tatrapayplus_api->initiatePayment(
    "redirect uri",
    $initiate_payment_request,
);
$response["object"]->getPaymentId();